### ISSUE TITLE
Babel 6 broke support for extending EventEmitter with super()

### ISSUE THREAD

**Issue by kripod** [babel/babel#2663]
C:\path\file.js:

``` js
import net from 'net';
import { EventEmitter } from 'events';
import BinarySerializer from './helpers/binary-serializer';
// import ...

export default class Connection extends EventEmitter {
	constructor(endpoint, joinKey, joinData, roomId) {
		super();

		this.isConnected = false;
		this.roomId = roomId;

		// ...
	}

	send(message) {
		this.sock.write(BinarySerializer.serializeMessage(message));
	}

	disconnect() {
		this.sock.close();
	}
}

```

.babelrc:

```
{
  "presets": ["es2015"]
}
```

Error:

```
SyntaxError: C:\path\file.js: 'this' is not allowed before super() (This is an error on an internal node. Probably an internal error)
    at File.buildCodeFrameError (C:\path\node_modules\babel-core\lib\transformation\file\index.js:408:15)
    at NodePath.buildCodeFrameError (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\path\index.js:135:26)
    at [object Object].ThisExpression (C:\path\node_modules\babel-preset-es2015\node_modules\babel-plugin-transform-es2015-classes\lib\vanilla.js:73:20)
    at NodePath._call (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\path\context.js:72:18)
    at NodePath.call (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\path\context.js:44:17)
    at NodePath.visit (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\path\context.js:102:12)
    at TraversalContext.visitQueue (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\context.js:151:16)
    at TraversalContext.visitMultiple (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\context.js:106:17)
    at TraversalContext.visit (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\context.js:193:19)
    at Function.traverse.node (C:\path\node_modules\babel-core\node_modules\babel-traverse\lib\index.js:139:17)
```

**Comment by safareli** [comment]
I have error error like this
Button.jsx : 
```
import React from 'react'
export default class Button extends React.Component {
  render() {
    return <button>foo</button>
  }
}
```

cli:
```
safareli$ babel-node Button.jsx 
/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/lib/transformation/file/index.js:519
      throw err;
      ^

SyntaxError: /Users/safareli/dev/mashbors/src/ui/objects/Button/Button.jsx: super() outside of class constructor (4:4)
  2 | export default class Button extends React.Component {
  3 |   Constructor(){
> 4 |     super()
    |     ^
  5 |   }
  6 |   render() {
  7 |     return <button></button>
    at Parser.pp.raise (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/parser/location.js:24:13)
    at Parser.pp.parseExprAtom (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/parser/expression.js:364:14)
    at Parser.<anonymous> (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/plugins/jsx/index.js:406:22)
    at Parser.<anonymous> (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/plugins/jsx/index.js:406:22)
    at Parser.<anonymous> (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/plugins/jsx/index.js:406:22)
    at Parser.parseExprAtom (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/plugins/jsx/index.js:406:22)
    at Parser.pp.parseExprSubscripts (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/parser/expression.js:244:19)
    at Parser.pp.parseMaybeUnary (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/parser/expression.js:225:19)
    at Parser.pp.parseExprOps (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/parser/expression.js:171:19)
    at Parser.pp.parseMaybeConditional (/usr/local/lib/node_modules/babel-cli/node_modules/babel-core/node_modules/babylon/lib/parser/expression.js:153:19)
```

.babelrc:
```
{
  "presets": ["es2015", "react"],
  "plugins": ["transform-decorators","transform-class-properties"]
}
```

dependencies:
```
    "babel-core": "^6.0.12",
    "babel-plugin-transform-class-properties": "^6.0.2",
    "babel-plugin-transform-decorators": "^6.0.2",
    "babel-preset-es2015": "^6.0.12",
    "babel-preset-react": "^6.0.12",
```

**Comment by sebmck** [comment]
@safareli Different problem and that's expected behaviour. You've capitalised your method, it's `constructor, not `Constructor`.

**Comment by safareli** [comment]
@sebmck  sorry i have copyed different log. have just updated it

**Comment by kidwm** [comment]
I've found that if I change from:
```js
export default class Button extends React.Component {
}
```

to

```js
class Button extends React.Component {
}
export default Button;
```

It will works with Babel 6

**Comment by thejameskyle** [comment]
@kidwm separate issue, see #2694

**Comment by AndrewRevinsky** [comment]
@kidwm :+1:

**Comment by rusel1989** [comment]
@kidwm :+1:

**Comment by sebmck** [comment]
I couldn't reproduce this with either of the examples provided. I've added them as regression tests though. Feel free to comment on this issue if it's still a problem.

**Comment by kripod** [comment]
I think this has been fixed along with #2694.
